pcsx2/DEV9/ATA/ATA_Info.cpp

```diff
@@ -34,8 +34,11 @@ void ATA::CreateHDDinfo(u64 sizeSectors)
 {
 	//PS2 is limited to 32bit size HDD (2TB), however,
 	//we don't yet support 48bit, so limit to 28bit size
-	constexpr u32 maxSize = (1 << 28) - 1; // 128Gb
-	sizeSectors = std::min<u32>(sizeSectors, maxSize);
+	if (!lba48Supported)
+	{
+		constexpr u32 maxSize = (1 << 28) - 1; // 128Gb
+		sizeSectors = std::min<u32>(sizeSectors, maxSize);
+	}
```

Comment:
Limit to 32bit sizes when `lba48Supported` is true, to match PS2 limits

Reply:
I am not sure where the 32-bit limit came from, but PS2 supports 48-bit fully (2Tb is 48-bit limit, not the 32.-bit limit). And if 48-bit are enabled - then we need to keep both limits: 28-bit and 48-bit.
![image](https://github.com/PCSX2/pcsx2/assets/621640/0e6873fe-ec7c-461d-8870-d3d3e2df8d32)

Reply:
Some quick maths shows that 32bit addresses can reach 2TB  `(512*2^32)/(1024*1024*1024*1024) = 2TB`,

The 32bit limit _was_ mentioned in your comments, I also seem to recall hearing that limit elsewhere as well, but a quick search shows that later versions of OPL can support much larger drives (of upto 144 Petabytes)  

And yes we would need limit each field separately

Reply:
Doh, my apologizes, ps2 really supports more than 2Tb, 2TB is the limit of APA partition scheme (or more particular - its MBR scheme), but does not directly reflects the ATA, so yes, ps2 can address 128PiB